### ISSUE TITLE
Unify PatchSchema and PatchCollection

### ISSUE THREAD

**Issue by AndrewSisley** [sourcenetwork/defradb#2401]
I think it would be nicer if there is just one entry point for this:

```
			testUtils.SchemaPatch{
				Patch: `
					[
						{ "op": "add", "path": "/Schema/Users/Fields/-", "value": {"Name": "name", "Kind": "String"} },
				                // Make the second collection the active one by moving its name from the first to the second
						{ "op": "move", "from": "/Collections/1/Name", "path": "/Collections/2/Name" }
					]
				`,
			},
```

This may be fairly fiddly to achieve however, for example in the above collection `2` does not exist yet.  Have a long thing about the dev-ex of this.

**Comment by fredcarle** [comment]
Since we are trying to separate the Schema and Collection within the code base, wouldn't unify this go against what has been done already?

**Comment by AndrewSisley** [comment]
> Since we are trying to separate the Schema and Collection within the code base, wouldn't unifying this go against what has been done already?

I dont think so, it is one end point, but they are very clearly separate paths (`Schema` vs `Collection`).  It just means users can submit many commands in a single patch (including test commands).

**Comment by fredcarle** [comment]
You're right. It's the same endpoint but the path is the differentiator. Thanks for pointing that out.
